package.json

```diff
@@ -92,5 +92,11 @@
   },
   "engines": {
     "node": ">=14"
+  },
+  "overrides": {
+    "react": "17.0.1"
```

Comment:
Maybe this has too far reaching impact. To me, this means that npm will force any dependency (or transitive dependency) to use react version `17.0.1`. 

The issue, AFAIK, is to do with react native testing library. I think then, scoping this override to that, would be better, such as:

```
{
  "overrides": {
    "baz": {
      "bar": {
        "foo": "1.0.0"
      }
    }
  }
}
```

this has the benefit of being explicit about the problem, and also reduces the impact. 

fyi, web and native has `react` marked as "external", so `react` will never be bundled with a widget or js action, and will rely on the version of `react` shipped with Spro client.

Whether we go with this or my proposed soliution will behave similarly, although I think the two advantages above are considerable.

Please let me know what you think.

https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

Reply:
Let's go with this for now, unless more packages impose similar issues.

Reply:
Thanks! I tried to apply your suggestion for the affected dependencies which I noticed when running `npm i` locally.